src/overlays/actors/ovl_Shot_Sun/z_shot_sun.c

```diff
@@ -26,17 +55,170 @@ const ActorInit Shot_Sun_InitVars = {
     (ActorFunc)ShotSun_Update,
     NULL,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Shot_Sun/ShotSun_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Shot_Sun/ShotSun_Destroy.s")
+static ColliderCylinderInit sCylinderInit = {
+    { COLTYPE_UNK10, 0x00, 0x09, 0x39, 0x10, COLSHAPE_CYLINDER },
+    { 0x00, { 0x00000000, 0x00, 0x00 }, { 0x00000020, 0x00, 0x00 }, 0x00, 0x01, 0x01 },
+    { 30, 60, 0, { 0, 0, 0 } },
+};
+
+void ShotSun_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ShotSun* this = THIS;
+    s32 temp_v0;
+
+    // Translation: Ocarina secret occurrence
+    osSyncPrintf("%d ---- オカリナの秘密発生!!!!!!!!!!!!!\n", this->actor.params);
+    temp_v0 = this->actor.params & 0xFF;
+    if (temp_v0 == 0x40 || temp_v0 == 0x41) {
+        this->unk_1A4[0] = 0;
+        this->actor.flags |= 0x10;
+        this->actor.flags |= 0x2000000;
+        this->actionFunc = &func_80BADF0C;
+        this->actor.flags |= 0x8000000;
+    } else {
+        Collider_InitCylinder(globalCtx, &this->collider);
+        Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+        this->actionFunc = &ShotSun_HyliaSunUpdate;
+        this->actor.flags &= ~1;
+    }
+}
+
+void ShotSun_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    ShotSun* this = THIS;
+    s32 temp_v0;
+
+    temp_v0 = this->actor.params & 0xFF;
```

Comment:
in general throughout this file, id probably name the temp "params". it doesnt seem the params is used for more than one purpose so i dont think you need to specify what kind of param it is.

not too important if you dont feel comfortable naming it right now though, just a suggestion

Reply:
Okay so I should name them params even though it's only the bottom byte? Also I named the fairy params fairyParams instead because that same function uses its own params as well.

Reply:
yea, generally params are bit packed and do a bunch of shifting to access different parts of the params. but in this case it doesnt, its only used for one purpose.
I guess ideally we could name it and call it a "type" or something-- 0x40 and 0x41 only make one small difference when calling the fairies. but i think "params" works at least for now